plugins/de.cognicrypt.staticanalyzer.kotlin/src/de/cognicrypt/staticanalyzer/kotlin/utilities/KotlinUtils.java

```diff
@@ -0,0 +1,261 @@
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IncrementalProjectBuilder;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.OperationCanceledException;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.widgets.MessageBox;
+import org.eclipse.ui.IWorkbenchWindow;
+import org.eclipse.ui.PlatformUI;
+import org.jetbrains.kotlin.core.compiler.KotlinCompiler;
+import org.jetbrains.kotlin.core.compiler.KotlinCompilerResult;
+import org.jetbrains.kotlin.core.model.KotlinNature;
+
+import de.cognicrypt.staticanalyzer.kotlin.Activator;
+
+public class KotlinUtils {
+	
+	public static void compileKotlinFiles(IProject ip) {
+		IJavaProject javaProject = JavaCore.create(ip);
+		
+		if(Platform.getBundle("org.jetbrains.kotlin.core") != null) {
+ 			if(KotlinNature.hasKotlinNature(ip)) {
+ 				try {
+ 					if(ip.hasNature("org.eclipse.m2e.core.maven2Nature")) {
+ 						if(KotlinUtils.verifyKotlinDependency(javaProject)) {
+ 							
+ 							System.out.println("Custom Builder started.");
+ 							final Job builder = new Job("Custom Builder") {
+								
+								@Override
+								protected IStatus run(IProgressMonitor monitor) {
+									try {
+										ip.build(IncrementalProjectBuilder.CLEAN_BUILD, null);
+									} catch (CoreException e) {
+										e.printStackTrace();
+									}
+									return Status.OK_STATUS;
+								}
+							};
+							builder.setPriority(Job.LONG);
+							builder.schedule();
+ 							
+ 							while(builder.getState() != Job.NONE) {
+								Thread.sleep(3);
+							}
+ 							System.out.println("Custom Builder finished.");
+ 			
+ 							KotlinUtils.waitForBuildAndRefreshJobs();
+ 							
+							KotlinCompilerResult result = KotlinCompiler.compileKotlinFiles(javaProject);
+							if(result.compiledCorrectly())
+								Activator.getDefault().logInfo("Finished compiling kotlin files.");
+							else
+								Activator.getDefault().logInfo("Cannot compile some kotlin files due to errors. Static analysis skipped them.");
+ 						}
+ 						else {
+ 							Activator.getDefault().logInfo("Cannot compile kotlin files without dependency.");
+ 							Activator.getDefault().logInfo("Static analysis skipped all kotlin files.");
+ 						}
+ 					}
+ 					else {
+ 						KotlinCompilerResult result = KotlinCompiler.compileKotlinFiles(javaProject);
```

Comment:
Same comments as above.
If I see this correctly, this branch is the same as the block above. If so, refactor both into one method.